src/PriorityQueue.php

```diff
@@ -337,19 +351,25 @@ public function hasPriority($priority)
      * Get the inner priority queue instance
      *
      * @throws Exception\DomainException
-     * @return SplPriorityQueue
+     * @return \SplPriorityQueue<TPriority, T>
```

Comment:
Can we perhaps import `SplPriorityQueue`? It got referenced everywhere with a prefix `\`

Reply:
I did try that - it was a nightmare. Because the class string can be replaced with any descendant of `\SplPriorityQueue` and the changes made to the Laminas version means that priorities get really messy because it uses a typed array for its priority internally. I think I probably tried about 3 or 4 times to get this looking sane - everything worked out better when the internal queue was typed as `\SplPriorityQueue<TPriority, T>` even though he Laminas version is used by default.

The ability to change the internal queue type should be dropped don't you think?

Reply:
> The ability to change the internal queue type should be dropped don't you think?

Not here, but yes: it's a useless feature.